das/metrics.go

```diff
@@ -110,29 +129,54 @@ func (d *DASer) InitMetrics() error {
 	return nil
 }
 
-func (m *metrics) observeSample(ctx context.Context, h *header.ExtendedHeader, sampleTime time.Duration, err error) {
+// observeSample records the time it took to sample a header +
+// the amount of sampled contiguous headers
+func (m *metrics) observeSample(
+	ctx context.Context,
+	h *header.ExtendedHeader,
+	sampleTime time.Duration,
+	err error,
+) {
 	if m == nil {
 		return
 	}
 	m.sampleTime.Record(ctx, sampleTime.Seconds(),
 		attribute.Bool("failed", err != nil),
-		attribute.Int("header_width", len(h.DAH.RowsRoots)))
+		attribute.Int("header_width", len(h.DAH.RowsRoots)),
+	)
+
 	m.sampled.Add(ctx, 1,
 		attribute.Bool("failed", err != nil),
-		attribute.Int("header_width", len(h.DAH.RowsRoots)))
+		attribute.Int("header_width", len(h.DAH.RowsRoots)),
+	)
+
 	atomic.StoreInt64(&m.lastSampledTS, time.Now().UTC().Unix())
+
+	if err == nil {
+		atomic.AddUint64(&m.totalSampledInt, 1)
+	}
 }
 
+// observeGetHeader records the time it took to get a header from the header store.
 func (m *metrics) observeGetHeader(ctx context.Context, d time.Duration) {
 	if m == nil {
 		return
 	}
 	m.getHeaderTime.Record(ctx, d.Seconds())
 }
 
+// observeNewHead records the network head.
 func (m *metrics) observeNewHead(ctx context.Context) {
 	if m == nil {
 		return
 	}
 	m.newHead.Add(ctx, 1)
 }
+
+// recordTotalSampled records the total sampled headers.
+func (m *metrics) recordTotalSampled(totalSampled uint64) {
+	if m == nil {
+		return
+	}
```

Comment:
```suggestion
```

please instead ensure that syncer never calls any method on a nil metrics.

Reply:
Do you mean `DASer`? We had a conversation me and @Wondertan about this pattern, and I mentioned that we should remove it.

For me to change this `if m == nil` would mean a major refactor to how we are doing metrics atm for DASer. The "better" pattern that we should have, that makes sense that metrics collection methods are not called in the first place when `Metrics` are disabled, is present in the blackbox metrics PR #1376.

To change this, we need a non-trivial refactor

Reply:
Referencing this in https://github.com/celestiaorg/celestia-node/issues/1705

Reply:
Since you already implemented the check here: https://github.com/celestiaorg/celestia-node/pull/1638/files#diff-a79f038a558af1c9917378c69183fb2563c3a0b6f49ae9464b1bfae68d2f1339R238-R240

we no longer need this check anyway